lib/ansible/modules/cloud/amazon/code_commit.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Shuang Wang <ooocamel@icloud.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = '''
+---
+module: code_commit
+version_added: "2.8"
+short_description: Represents AWS CodeCommit
+description: Creates,deletes repository in AWS CodeCommit
+author: Shuang Wang (@ptux)
+
+requirements:
+  - botocore
+  - boto3
+  - python >= 2.6
+
+options:
+  name:
+    description:
+      - name of repository.
+    required: true
+  comment:
+    description:
+      - description or comment of repository.
+    required: false
+  state:
+    description:
+      - Specifies the state of repository.
+    required: true
+    choices: [ 'present', 'absent' ]
+
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+RETURN = '''
+results:
+    description: A complex type that contains information about results
+    returned: changed
+    type: dict
+
+'''
+
+EXAMPLES = '''
+# Create new A repository
+- code_commit:
+    name: repo
+    state: present
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    boto_exception,
+    ec2_argument_spec
+)
+
+
+class CodeCommit(object):
+    def __init__(self, module=None):
+        self._module = module
+        self._connection = self._module.client('codecommit')
+        self._check_mode = self._module.check_mode
+
+    def process(self):
+        results = dict(changed=False)
+        changed = False
+        repository_exists = self._repository_exists()
+
+        if self._module.params['state'] == 'present' and not repository_exists:
+            changed = True
+            if not self._module.check_mode:
+                changed, results = self._create_repository()
+        if self._module.params['state'] == 'absent' and repository_exists:
+            changed = True
+            if not self._module.check_mode:
+                changed, results = self._delete_repository()
+        return changed, results
+
+    def _repository_exists(self):
+        try:
+            repository_exists = False
+            repositories = self._connection.list_repositories()['repositories']
```

Comment:
This call should probably be paginated: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/codecommit.html#CodeCommit.Paginator.ListRepositories